### ISSUE TITLE
create-env with trusted certificates for resulting VM

### ISSUE THREAD

**Issue by Kiemes** [cloudfoundry/bosh#1718]
We would like to deploy a director using `create-env` and hand in some additional trusted certificates via `bosh.yml` manifest. The result should be that the director trusts servers which are signed by this CA. I guess the place to put those additional certs would be the system trust store. Is there a way to do it we are not aware of or is it not possible?

We did find https://bosh.io/docs/trusted-certs.html but this only adds certificates to the VMs deployed with the director not the director VM itself. The corresponding "[update code](https://github.com/cloudfoundry/bosh-agent/blob/1a6b1e11acd941e65c4f4155c22ff9a8f76098f9/platform/cert/cert_manager.go#L19-L33)" is in the agent.

This code could be reused if `create-env` would read the mentioned `trusted_certs` section of the manifest and write it in the settings.json which will be read by the agent.

**Comment by cppforlife** [comment]
@Kiemes we've added ca_certs job (https://bosh.io/jobs/ca_certs?source=github.com/cloudfoundry/os-conf-release&version=12) to aid that. i was even wondering if we should just replace native trusted_certs funcationality with this job, once we have ordering figured out.

**Comment by manno** [comment]
Are BOSH instance groups applied in the order that is given in the manifest? The `bosh create-env` might depend on the custom certificate.
If that works it would be nice to remove [native trusted certs](https://bosh.io/docs/trusted-certs.html) from the bosh release.

**Comment by Kiemes** [comment]
I now had time to try out the ca_certs job of os-conf.
Thinking about the process, I wonder what happens if you need the additional certs during compilation time? In the create-env case, this is done on the director VM, isn't is? Jobs will be applied after compilation, right?

The agent picking up settings.json and handling the certs would happen before compilation I guess.

**Comment by cppforlife** [comment]
@Kiemes for blobstore access?

**Comment by Kiemes** [comment]
@cppforlife yes, for example. Or for the compiled packages cache. If you have an own S3 compatible system with your own certificates.
